metrics/src/main/java/org/aerogear/mobile/metrics/interfaces/MetricsProvider.java

```diff
@@ -0,0 +1,40 @@
+package org.aerogear.mobile.metrics.interfaces;
+
+import android.content.Context;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Objects;
+
+public abstract class MetricsProvider {
+    public String namespace() {
```

Comment:
From what I've understand this method should be abstract. And I suggest it returns an enum `MetricsProviderNamespace` that references all existing namespaces instead of a hardcode string.